### DISCUSSION TITLE
Text input for Field

### DISCUSSION THREAD

**Discussion by Balt2** [react-querybuilder/react-querybuilder#347] [Q&A] [answered]
Is it possible to have a Input text field instead of a select? This would make the query builder extremely configurable.

**Comment by jakeboone02** [comment, marked as answer]
Hi, @Balt2! Yes, you can replace any component within the query builder with your own implementation. [This codesandbox](https://codesandbox.io/s/react-querybuilder-text-input-field-selector-7vvke6?file=/src/App.tsx:470-657) uses a text input as the field selector.

This is the custom component:

```tsx
const FieldSelector = (props: FieldSelectorProps) => (
  <input
    value={props.value}
    onChange={(e) => props.handleOnChange(e.target.value)}
    className={props.className}
  />
);
```

To use it, assign it to the `fieldSelector` property of the `controlElements` prop, like this:

```tsx
export const App = () => {
  const [query, setQuery] = useState(initialQuery);

  return (
    <div>
      <QueryBuilder
        fields={fields}
        query={query}
        onQueryChange={(q) => setQuery(q)}
        controlElements={{
          fieldSelector: FieldSelector
        }}
      />
  );
};
```

**Reply by jakeboone02** [reply]
Hi @Balt2. If the response above helped you out, could you mark it as the answer?

**Reply by Balt2** [reply]
Yes, thank you! This helped a ton.
